internal/driver/luks.go

```diff
@@ -122,20 +132,23 @@ func luksFormat(ctx LuksContext, source string) error {
 
 	klog.V(4).Info("executing cryptsetup luksFormat command ", cryptsetupArgs)
 
-	cmd := exec.Command(cryptsetupCmd, cryptsetupArgs...)
+	cmd := e.Exec.Command(cryptsetupCmd, cryptsetupArgs...)
 
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		return fmt.Errorf("failed to get stdin pipe for cryptsetup, got err: %w", err)
-	}
+	// Set the encryption key as stdin
+	cmd.SetStdin(strings.NewReader(ctx.EncryptionKey))
 
-	if _, err := io.WriteString(stdin, ctx.EncryptionKey); err != nil {
-		return fmt.Errorf("failed to write to stdin pipe for cryptsetup, got err: %w, closing pipe: %w", err, stdin.Close())
-	}
+	// stdin, err := cmd.StdinPipe()
```

Comment:
we should remove commented-out code here or describe why it needs to stay commented out, a comment describing what functionality we might want to add in later is likely a better approach

Reply:
I'm leaving it here until Luks E2E tests are merged in. Its to make sure the new code works and didn't break luks functionality.